### ISSUE TITLE
[BUG]Creating .NET BlobClient causes FileNotFoundException for System.Diagnostics.DiagnosticSource Version=4.0.4.0 in Release configuration.

### ISSUE THREAD

**Issue by brewerpm** [Azure/azure-sdk-for-net#36454]
### Library name and version

Azure.Storage.Blobs 12.16.0

### Describe the bug

I have a hybrid solution (Native C++, C++/CLI, c#, .NET - 15 projects in the solution) using up-to-date Visual Studio 2022. My solution targets framework version 4.7.2. I have two c# class library projects each of which has functions that upload or download files from Azure blob storage. As such, both projects use the Azure.Storage.Blobs NuGet package latest version 12.16.0.

When I run the program in debug mode everything works fine. But in Release config, when either of the two functions (they're in different C# projects) attempts to create a BlobClient  I get the following exception:
	System.IO.FileNotFoundException: 'Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=4.0.4.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.

According to NuGet.org, the dependency tree for Azure.Storage.Blobs->Azure.Storage.Common 12.15.0->Azure.Core (>= 1.31.0)->System.Diagnostics.DiagnosticSource (>= 4.6.0)->System.Memory (>= 4.5.5)->System.Buffers (>= 4.5.1).
I have the latest stable version of all packages installed for both projects including System.Diagnostics.DiagnosticsSource NuGet package (7.0.2). The 7.0.2 System.Diagnostics.DiagnosticsSource package contains runtime version 4.0.30319.

What is causing this exception and why would it be looking for System.Diagnostics.DiagnosticSource Version=4.0.4.0?

Note that I have a much simpler solution that uses exactly the same code to upload & download Azure blobs which DOES NOT get this FileNotFoundException. In other words, there is no way I can replicate this problem with a simpler solution. I have tried to locate System.Diagnostics.DiagnosticSource, Version=4.0.4.0, but it does not seem to exist.

I would be happy to supply the app.config files for the two projects but thought it might be too must stuff for now.

### Expected behavior

Uploads the file to blob storage.

### Actual behavior

System.IO.FileNotFoundException: 'Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=4.0.4.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.

### Reproduction Steps

Below is the code I'm using. As stated above, I have a simple c# project/solution that uses the same code **but does not get the exception**; therefore, I see no way to easily reproduce.

public bool UploadFileToAzureBlobStorage(string Path, string containerName, string BlobName)
{
    string userName = System.Security.Principal.WindowsIdentity.GetCurrent().Name;
    EnvironmentVariableTarget target;
    if (userName == @"NT AUTHORITY\SYSTEM")
		target = EnvironmentVariableTarget.Machine;
    else
		target = EnvironmentVariableTarget.User;
    string connectionString = Environment.GetEnvironmentVariable("HOROSCOPE_STORAGE_ACCT_CONNECTION_STRING", target);
    //string containerName = "chart-reports";
    string blobName = BlobName;
    string filePath = Path;

    BlobContainerClient container = null;
    BlobClient blob = null;
    try
    {
		container = new BlobContainerClient(connectionString, containerName);
		blob = container.GetBlobClient(blobName);
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.Message);
        return false;
    }
    // Upload local file
    try
    {
        // set content-type from https://github.com/Azure/azure-sdk-for-net/issues/9823
        blob.Upload(filePath, new BlobHttpHeaders {ContentType = "text/html"},
            conditions: null); // "conditions: null" means overwrite
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.Message);
        return false;
    }
    return true;
}

### Environment

Visual Studio 2022 17.5.5
.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.203\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

**Comment by brewerpm** [comment]
Turns out that the DLL in question (System.Diagnostics.DiagnosticSource) was not being copied to the output folder even though Copy Local was set. I should have realized this right away but was confused by the fact that the system kept looking for v4.0.4.0 instead of the v7.0.0.2 that was referenced in the two projects (including the app.config, packages.config, etc.) This turns out to be a common problem when building with VS or MSBuild (according to many posts.)
What happens is that there is a main, startup project, A, and two other projects, B and C, on which A depends. Both B and C call X.dll. X.dll depends on Y.dll but, even though Y.dll is specifically refenced in both B and C projects, the build process, in trying to be tidy, does not copy Y.dll to the output folder. One way to fix this (the way I did it) is to add the reference to project A. Then it for sure gets copied to the output folder.

This is a fairly common problem with more complex solutions. See, for example, [c# - Dependent DLL is not getting copied to the build output folder in Visual Studio - Stack Overflow.](https://stackoverflow.com/questions/15816769/dependent-dll-is-not-getting-copied-to-the-build-output-folder-in-visual-studio)
